distribution/push_v2.go

```diff
@@ -425,26 +417,17 @@ func (pd *v2PushDescriptor) uploadUsingSession(
 	diffID layer.DiffID,
 	layerUpload distribution.BlobWriter,
 ) (distribution.Descriptor, error) {
-	arch, err := pd.layer.TarStream()
+	reader, release, err := pd.layer.Content(ctx, progressOutput, pd.ID(), schema2.MediaTypeLayer)
```

Comment:
ideally progress output would be done on the returned reader but we are returning a compressed stream but reporting progress on the uncompressed stream. Maybe it would be better to just have a compression flag  and let it return an `io.ReadCloser` and `int64` size for progress.